### PR TITLE
Added certbot plugin for Beget DNS service

### PR DESCRIPTION
**Pull request by Medan-rfz**
Added support for DNS provider Beget using the sertbot plugin certbot-dns-beget

### CODE REVIEW

**Comment by jc21** [comment]
Please test this docker image and let me know if the DNS plugin change works as expected. Thanks

**Comment by zagorodnovv** [comment]
I used the NPM image on 01.12.2024 nginxproxymanager/nginx-proxy-manager-dev:pr-4148

When logging into SSL Certificates - And Let's Encrypt Certificate - Use a DNS Chellenge, there is nothing related to Beget in the list

**Comment by jc21** [comment]
It shows up for me:

Screenshot 2024-12-02 at 10 41 55 am

**Comment by zagorodnovv** [comment]
Greetings again!

You are right, it was a mistake on my part when I said that there is no Beget in the list. I started NPM in another browser and now I see this entry.

Everything worked for me, I want to tell you about the important points:

1) The website for the Beget API says:

By default, API access is not provided.
To work with the API, you need to set a separate password to access the API and allow API authentication in the control panel (https://cp.beget.com/settings/access/api).

That is, first you need to go to your personal account Beget.com enable this feature.

2) There is a flaw in the Credentials File Content field, it is written like this and it does not work:

# Beget API credentials used by Certbot
beget_plugin_username = usernameeget_plugin_password = password

I did so and it worked:

# Beget API credentials used by Certbot
beget_plugin_username = my_username_beget
beget_plugin_password = my_API_password_beget

3) When trying to create an SSL certificate, I had 2 or 3 unsuccessful attempts, as a result, I set the value of Propagation Seconds to 60

P.S.

I myself use Nginx Proxy Manager LXC (https://community-scripts.github.io/ProxmoxVE/scripts?id=nginxproxymanager ), there is no way to use Beget. How soon can I start using the Beget plugin?

**Comment by zagorodnovv** [comment]
Greetings again!

You are right, it was a mistake on my part when I said that there is no Beget in the list. I started NPM in another browser and now I see this entry.

Everything worked for me, I want to tell you about the important points:

1) The website for the Beget API says:

By default, API access is not provided.
To work with the API, you need to set a separate password to access the API and allow API authentication in the control panel (https://cp.beget.com/settings/access/api).

That is, first you need to go to your personal account Beget.com enable this feature.

2) There is a flaw in the Credentials File Content field, it is written like this and it does not work:

```
# Beget API credentials used by Certbot
beget_plugin_username = usernameeget_plugin_password = password
```

I did so and it worked:

```
# Beget API credentials used by Certbot
beget_plugin_username = my_username_beget
beget_plugin_password = my_API_password_beget
```

3) When trying to create an SSL certificate, I had 2 or 3 unsuccessful attempts, as a result, I set the value of `Propagation Seconds to 60`

P.S.

I myself use Nginx Proxy Manager LXC (https://community-scripts.github.io/ProxmoxVE/scripts?id=nginxproxymanager), there is no way to use Beget. How soon can I start using the Beget plugin?

File: global/certbot-dns-plugins.json [hunk 1]
```diff
@@ -31,6 +31,14 @@
...
 		"full_plugin_name": "dns-azure"
 	},
+	"beget": {
+		"name":"Beget",
+		"package_name": "certbot-beget-plugin",
+		"version": "~=1.0.0.dev8",
+		"dependencies": "",
+		"credentials": "# Beget API credentials used by Certbot\nbeget_plugin_username = username\beget_plugin_password = password",
```
Comment (jc21) [inline global/certbot-dns-plugins.json:39]:
```
...beget_plugin_username\beg...
```
should be
```
...beget_plugin_username\nbeg...
```
it's missing the `n` after the slash

Comment (Medan-rfz) [reply global/certbot-dns-plugins.json]: I apologize for the inattention, I made a correction

**Pull request merged**